bigquery_etl/cli/query.py

```diff
@@ -1262,6 +1265,151 @@ def render(name, sql_dir, output_dir):
             click.echo(rendered_sql)
 
 
+def _parse_partition_date(partition_date):
+    params = partition_date.split(":")
+    if len(params) != 3:
+        return None
+
+    # Check date format
+    try:
+        datetime.datetime.strptime(params[2], "%Y-%m-%d").date()
+    except ValueError:
+        return None
+
+    # Check column name
+    if re.match(r"^\w+$", params[0]):
+        return {params[0]: params[2]}
+
+
+def _validate_partition_date(ctx, param, partition_date):
+    """Process the CLI parameter check_date and set the parameter for BigQuery."""
+    # Will be None if launched from Airflow.  Also ctx.args is not populated at this stage.
+    if partition_date is None:
+        return None
+    else:
+        parsed = _parse_partition_date(partition_date)
+        if parsed is None:
+            raise click.BadParameter("Format must be <column-name>::<yyyy-mm-dd>")
+        return parsed
```

Comment:
Generally, if we want we could perhaps do something like this instead and outsource the validation logic to `click`:
https://github.com/mozilla/bigquery-etl/blob/main/bigquery_etl/cli/backfill.py#L56-L63

Reply:
This function is already specified as the validation callback for the partition option.